### ISSUE TITLE
Error while adding revisions

### ISSUE THREAD

**Issue by odisseo79** [danmunn/redmine_dmsf#1239]
Dear all,
I am having a problem when adding a revision to a document.
The error I get is "Operation not permitted @ apply2files - /media/WinShare/dmsf/2021/04/210420134032_464_TES-xx-xx-713_01_BOM.xlsx".
I can upload files without any problem, but then I want to update a revision I get this error.
When I look in the folder when the documents are uploaded, I can find the new file. However I cannot reach it from the redmine website.
In the following the production log from redmine.

`Started POST "/dmsf/files/464/revision/create" for 10.100.2.101 at 2021-04-20 13:40:32 +0000
Processing by DmsfFilesController#create_revision as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"g5N7B7DjhzWbxbzyQbC+kzt1b9K2n+y16ocUm7W8a4bD9chrG8/hgG1t1xzUlXFfZJMR/m7YJtfgPkF67A3qsQ==", "dmsf_file_revision"=>{"title"=>"TES-20-003-713 01 BOM", "name"=>"TES-xx-xxx-713_01_BOM.xlsx", "description"=>"Bill of Material", "comment"=>""}, "version"=>"1", "custom_version_major"=>"3", "custom_version_minor"=>"1", "dmsf_attachments"=>{"1"=>{"filename"=>"TES-xx-xxx-713_01_BOM.xlsx", "token"=>"664.d539c68f631fe2e7d91adcfd0d7e32e3d4fa2828fab04ef3c3eb249f081b5fcd"}}, "commit"=>"Crea", "id"=>"464"}
  Current user: gxxx.xxxxxx (id=5)
Operation not permitted @ apply2files - /media/WinShare/dmsf/2021/04/210420134032_464_TES-xx-xx-713_01_BOM.xlsx
Redirected to http://10.100.2.100/dmsf/files/464`

Regards
Guido

**Comment by picman** [comment]
It seems to be caused by insufficient filesystem permissions. Check whether the web server user has got R/W permissions to the given folder.

**Comment by odisseo79** [comment]
Solved, but I do not know ifin the correct manner.
I am using a Windows Shared folder mounted at start-up.
In the mounting instruction I also specified ownership information: I had the problem with bitnami and www-data users.
The only way I do not have problems is assigning the ownership to daemon. I deduced this by using a different folder on the virtual machine and verifying the ownership of the uploaded files.
